osgar/drivers/logzeromq.py

```diff
@@ -0,0 +1,57 @@
+"""
+  Logger for ZeroMQ communication
+"""
+
+from threading import Thread
+
+import zmq
+
+from osgar.bus import BusShutdownException
+
+
+class LogZeroMQ:
+    def __init__(self, config, bus):
+        mode = config['mode']
+        endpoint = config['endpoint']
+
+        context = zmq.Context()
+        print("Connecting to hello world server ...")
+        if mode == 'PULL':
+            self.socket = context.socket(zmq.PULL)
+            self.thread = Thread(target=self.run_input, daemon=True)
+        elif mode == 'PUSH':
+            self.socket = context.socket(zmq.PUSH)
+            self.thread = Thread(target=self.run_output, daemon=True)
+        else:
+            assert False, mode  # unknown/unsupported mode
+
+        self.socket.connect(endpoint)  # TODO config
+
+        self.bus = bus
+
+    def start(self):
+        self.thread.start()
+
+    def join(self, timeout=None):
+        self.thread.join(timeout=timeout)
+
+    def run_input(self):
+        while self.bus.is_alive():
+            message = self.socket.recv()
```

Comment:
Can there be a timeout here? It seems that if it can block forever in recv, that would prevent the thread from noticing that `self.bus.is_alive()` is false.